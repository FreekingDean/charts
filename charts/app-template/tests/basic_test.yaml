tests:
  # or you can snapshot the whole manifest
  - it: manifest should match snapshot
    asserts:
      - matchSnapshot: {}
