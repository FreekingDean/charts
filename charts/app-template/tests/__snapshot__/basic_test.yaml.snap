manifest should match snapshot:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        reloader.stakater.com/auto: "true"
      labels:
        app.kubernetes.io/component: RELEASE-NAME
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        helm.sh/chart: app-template-0.0.11
      name: RELEASE-NAME
    spec:
      replicas: 2
      revisionHistoryLimit: 3
      selector:
        matchLabels:
          app.kubernetes.io/component: RELEASE-NAME
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: RELEASE-NAME
      strategy:
        type: RollingUpdate
      template:
        metadata:
          labels:
            app.kubernetes.io/component: RELEASE-NAME
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/name: RELEASE-NAME
        spec:
          automountServiceAccountToken: true
          containers:
            - image: busybox:latest
              livenessProbe:
                failureThreshold: 3
                initialDelaySeconds: 0
                periodSeconds: 10
                tcpSocket:
                  port: 8080
                timeoutSeconds: 1
              name: RELEASE-NAME
              readinessProbe:
                failureThreshold: 3
                initialDelaySeconds: 0
                periodSeconds: 10
                tcpSocket:
                  port: 8080
                timeoutSeconds: 1
              startupProbe:
                failureThreshold: 30
                initialDelaySeconds: 0
                periodSeconds: 5
                tcpSocket:
                  port: 8080
                timeoutSeconds: 1
          dnsPolicy: ClusterFirst
          enableServiceLinks: false
          hostIPC: false
          hostNetwork: false
          hostPID: false
          serviceAccountName: default
  2: |
    apiVersion: v1
    kind: Service
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        app.kubernetes.io/service: RELEASE-NAME
        helm.sh/chart: app-template-0.0.11
      name: RELEASE-NAME
    spec:
      ports:
        - name: http
          port: 8080
          protocol: TCP
          targetPort: 8080
      selector:
        app.kubernetes.io/component: RELEASE-NAME
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/name: RELEASE-NAME
      type: ClusterIP
  3: |
    apiVersion: networking.k8s.io/v1
    kind: Ingress
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        helm.sh/chart: app-template-0.0.11
      name: RELEASE-NAME
    spec:
      ingressClassName: traefik
      rules:
        - host: .deangalvin.dev
          http:
            paths:
              - backend:
                  service:
                    name: RELEASE-NAME
                    port:
                      number: 8080
                path: /
                pathType: Prefix
      tls:
        - hosts:
            - .deangalvin.dev
          secretName: deangalvin-dev-production-tls
  4: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: RELEASE-NAME
        helm.sh/chart: app-template-0.0.11
      name: RELEASE-NAME
    spec:
      endpoints:
        - interval: 1m
          path: /metrics
          port: http
          scheme: http
          scrapeTimeout: 10s
      jobLabel: RELEASE-NAME
      namespaceSelector:
        matchNames:
          - NAMESPACE
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: RELEASE-NAME
          app.kubernetes.io/service: RELEASE-NAME
